tests/integration/targets/vmware_guest_instant_clone/tasks/main.yml

```diff
@@ -0,0 +1,387 @@
+
+- when: vcsim is not defined
+  block:
+  - name: Instant Clone a VM with parent VM
+    vmware_guest_instant_clone:
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      validate_certs: false
+      folder: "{{ f0 }}"
+      datastore: "{{ rw_datastore }}"
+      datacenter: "{{ dc1 }}"
+      host: "{{ esxi1 }}"
+      name: "cloned_vm_from_vm"
+      parent_vm: "test_vm1"
+      resource_pool: Resources
+    register: instant_clone_from_vm_parent
+
+  - name: assert that changes were made
+    assert:
+      that:
+          - instant_clone_from_vm_parent is changed
+
+  - name: set state to poweroff the Cloned VM
+    vmware_guest_powerstate:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: "cloned_vm_from_vm"
+      folder: "{{ f0 }}"
+      state: powered-off
+    register: poweroff_instant_clone_from_vm_parent
+
+  - debug:
+      var: poweroff_instant_clone_from_vm_parent
+
+  - name: Clean VM
+    vmware_guest:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: "cloned_vm_from_vm"
+      datacenter: "{{ dc1 }}"
+      state: absent
+    register: delete_instant_clone_from_vm_parent
+    ignore_errors: true
+
+  - debug: var=delete_instant_clone_from_vm_parent
+
+  - name: Instant Clone a VM when datastore cluster is specified
+    vmware_guest_instant_clone:
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      validate_certs: false
+      folder: "{{ f0 }}"
+      vcenter_port: '{{ vcenter_port }}'
+      datastore: "{{ rw_datastore }}"
+      datacenter: "{{ dc1 }}"
+      host: "{{ esxi1 }}"
+      name: "cloned_vm_from_vm_cluster"
+      parent_vm: "test_vm1"
+      resource_pool: test_resource_001
+    register: instant_clone_from_vm_when_cluster_is_specified
+
+  - name: assert that changes were made
+    assert:
+      that:
+          - instant_clone_from_vm_when_cluster_is_specified is changed
+
+  - name: set state to poweroff the Cloned VM
+    vmware_guest_powerstate:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: "cloned_vm_from_vm_cluster"
+      folder: "{{ f0 }}"
+      state: powered-off
+    register: poweroff_instant_clone_from_vm_when_cluster
+
+  - debug:
+      var: poweroff_instant_clone_from_vm_when_cluster
+
+  - name: Clean VM
+    vmware_guest:
+      validate_certs: false
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      name: "cloned_vm_from_vm_cluster"
+      datacenter: "{{ dc1 }}"
+      state: absent
```

Comment:
I seem that the vmware_guest module needs when deleting an instant clone VM because the vmware_guest_instance_clone module hasn’t a state parameter.

In other modules as necessary has the state parameter.
So I also think this new module needs its parameter.

What do you think? @Akasurde

Reply:
Could you please confirm this @Akasurde?

Reply:
Can you please add an explanation about the vmware_guest is necessary when removing a VM in notes?
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-fields

And can you please also add a removing example in the EXAMPLE block?

Reply:
Ok

Reply:
Done can you check anything else?